src/pages/Home.jsx

```diff
@@ -1,41 +1,58 @@
 import { useEffect, useState } from "react";
 import SingleBook from "../components/SingleBook";
 import { Link } from "react-router-dom";
+import GenreFilter from "../components/GenreFilter";
 
 export default function Home({claimed}) {
     const [bookInfo, setBookInfo] = useState([])
+    const [filteredBooks, setFilteredBooks] = useState([])
+    const [genres, setGenres] = useState([])
+    const [selectedGenre, setSelectedGenre] = useState("")
 
     function getData() {
         const url = `https://book-swap-api.dev.io-academy.uk/api/books?claimed=${claimed}`
-        // if ({claimed}=1){
-        //    url = "https://book-swap-api.dev.io-academy.uk/api/books?claimed=1"
-        // }
-        // else {
-        //     url = "https://book-swap-api.dev.io-academy.uk/api/books?claimed=0"
-        // }
 
         fetch(url)
          .then(res => res.json())
          .then(fetchedInfo => {
-            setBookInfo(fetchedInfo.data)
+            setBookInfo(fetchedInfo.data);
+            setFilteredBooks(fetchedInfo.data) 
+
+            const uniqueGenres = Array.from (new Set(fetchedInfo.data.map(book => book.genre.name)));
+            setGenres(uniqueGenres)
          })
+         .catch(error => console.error("Error fetching books", error))
     }
 
     useEffect(getData, [claimed])
 
+    function handleFilterChange(genre) {
+      setSelectedGenre(genre)
+      if (genre === "") {
+        setFilteredBooks(bookInfo)
+      } else {
+        setFilteredBooks(bookInfo.filter(book => book.genre.name === genre))
```

Comment:
Similar story - the get all books route in the API supports automatic filtering by genre. Same as before, if it didn't, this would be a great way to do it.

Generally speaking, you want to get the API to do as much work for you as possible, only dealing with data in react when it's absolutely required (like stuff they type into a form etc)